File: kafka-nodejs-demo/vpc/consumer.js

const Kafka = require('node-rdkafka');
const config = require('./setting');
console.log(Kafka.features);
console.log(Kafka.librdkafkaVersion);
console.log(config)

var consumer = new Kafka.KafkaConsumer({
	/*'debug': 'all',*/ 
    'api.version.request': 'true',
	'bootstrap.servers': config['bootstrap_servers'],
    'group.id' : config['consumer_id'],
    'heartbeat.interval.ms': 3000,
    'session.timeout.ms': 30000,
    'max.poll.interval.ms': 40000,
    'fetch.max.bytes': 1024000,
    'max.partition.fetch.bytes': 256000
});


// Flowing mode:
consumer.connect();

consumer.on('ready', function() {
  console.log("connect ok");

  consumer.subscribe([config['topic_name']]);

  // Consume from the librdtesting-01 topic. This is what determines
  // the mode we are running in. By not specifying a callback (or specifying
  // only a callback) we get messages as soon as they are available.
  consumer.consume();
})

consumer.on('data', function(data) {
  // Output the actual message contents
  console.log(data);
});


consumer.on('event.log', function(event) {
      console.log("event.log", event);
});

consumer.on('error', function(error) {
	console.log("error:" + error);
});

consumer.on('event', function(event) {
        console.log("event:" + event);
});

File: kafka-confluent-go-demo/vpc-9092/consumer/consumer.go
package main

import (
	"fmt"
    "kafkagodemo/configs"
    "github.com/confluentinc/confluent-kafka-go/kafka"
)

var cfg *configs.KafkaConfig
var consumer  *kafka.Consumer

func doInit() {

	fmt.Print("init kafka producer, it may take a few seconds to init the connection\n")

	var err error

	cfg = &configs.KafkaConfig{}
	configs.LoadJsonConfig(cfg, "kafka.json")


    consumer, err = kafka.NewConsumer(&kafka.ConfigMap{
		"bootstrap.servers": cfg.Servers,
		"group.id":          cfg.ConsumerId,
		"auto.offset.reset": "latest",
		"api.version.request": "true",
		"heartbeat.interval.ms": 3000,
		"session.timeout.ms": 30000,
		"max.poll.interval.ms": 120000,
		"fetch.max.bytes": 1024000,
		"max.partition.fetch.bytes": 256000,
	})
	if err != nil {
		panic(err)
	}

}


func main() {

	//Init it once
	doInit()

	consumer.SubscribeTopics([]string{cfg.Topic}, nil)

	for {
		msg, err := consumer.ReadMessage(-1)
		if err == nil {
			fmt.Printf("Message on %s: %s\n", msg.TopicPartition, string(msg.Value))
		} else {
			// The client will automatically try to recover from all errors.
			fmt.Printf("Consumer error: %v (%v)\n", err, msg)
		}
	}

	consumer.Close()
}